jetty-io/src/main/java/org/eclipse/jetty/io/ChannelEndPoint.java

```diff
@@ -357,12 +353,16 @@ public Runnable onSelected()
         return task;
     }
 
+    private void updateKey(Selector selector)
```

Comment:
maybe call this one `updateKeyAction`

Reply:
Not sure I agree. This method updates the key, it's not the action. The field is called `_updateKeyAction` because it's the action that it is submitted to the selector.

Reply:
So the updateKeyAction method is submitted to the selector, which calls us back here and then we call compute on the selector because our _key might be out of date....  All to avoid an array deref in `keyFor`  !!! I think we are getting too tricky.

The other callback pass the key, so we can just use it.  This callback passes the selector, so let's use use `keyFor` here and get rid of `_key` field.